src/services/DhService.ts

```diff
@@ -259,41 +279,45 @@ export abstract class DhService<TDH, TClient>
-        // If selectionOnly is true, the line number in the error will be
-        // relative to the selection (Python line numbers are 1 based. vscode
-        // line numbers are zero based.)
-        const fileLine =
-          (selectionOnly ? line + editor.selection.start.line : line) - 1;
-
-        // There seems to be an error for certain Python versions where line
-        // numbers are shown as -1. In such cases, we'll just mark the first
-        // token on the first line to at least flag the file as having an error.
-        const startLine = Math.max(0, fileLine);
-
-        // Zero length will flag a token instead of a line
-        const lineLength =
-          fileLine < 0 ? 0 : editor.document.lineAt(fileLine).text.length;
-
-        // Diagnostic representing the line of code that produced the server error
-        const diagnostic: vscode.Diagnostic = {
-          message: value == null ? error : `${value}\n${error}`,
-          severity: vscode.DiagnosticSeverity.Error,
-          range: new vscode.Range(startLine, 0, startLine, lineLength),
-          source: 'deephaven',
-        };
-
-        this.diagnosticsCollection.set(editor.document.uri, [diagnostic]);
+      if (editor.document.languageId === 'python') {
+        const { line, value } = parseServerError(error);
+
+        if (line != null) {
+          // If selectionOnly is true, the line number in the error will be
+          // relative to the selection (Python line numbers are 1 based. vscode
+          // line numbers are zero based.)
+          const fileLine =
+            (selectionOnly ? line + editor.selection.start.line : line) - 1;
+
+          // There seems to be an error for certain Python versions where line
+          // numbers are shown as -1. In such cases, we'll just mark the first
+          // token on the first line to at least flag the file as having an error.
+          const startLine = Math.max(0, fileLine);
+
+          // Zero length will flag a token instead of a line
+          const lineLength =
+            fileLine < 0 ? 0 : editor.document.lineAt(fileLine).text.length;
+
+          // Diagnostic representing the line of code that produced the server error
+          const diagnostic: vscode.Diagnostic = {
+            message: value == null ? error : `${value}\n${error}`,
+            severity: vscode.DiagnosticSeverity.Error,
+            range: new vscode.Range(startLine, 0, startLine, lineLength),
+            source: 'deephaven',
+          };
+
+          this.diagnosticsCollection.set(editor.document.uri, [diagnostic]);
+        }
       }
 
       return;
     }
 
     const changed = [...result!.changes.created, ...result!.changes.updated];
 
-    let lastPanel: vscode.WebviewPanel | null = null;
+    // Have to set this with type assertion since TypeScript can't figure out
+    // assignments inside of the `forEach` and will treat `lastPanel` as `null`.
+    let lastPanel = null as vscode.WebviewPanel | null;
```

Comment:
Kind of annoying, weird that TypeScript does that.

Reply:
@mofojed  For posterity, turns out this is only an issue since the variable gets set in a `forEach` callback. It works fine without the type assertion inside of a `for..of` loop.